### PR TITLE
Negative test for cutom taints

### PR DESCRIPTION
**Pull request by vkathole**

### CODE REVIEW

File: tests/functional/z_cluster/nodes/test_non_ocs_taint_and_toleration.py [hunk 1]
```diff
@@ -227,3 +228,104 @@ def test_non_ocs_taint_and_tolerations(self):
...
+    def test_custom_taint_node_operations(self, nodes):
+        """
+        Test runs the following steps
+        1. Add toleration in storagecluster CR and odf-operator subscription.
+        2. Verify toleration applied in ODF subscription is reflecting
+           for other subscriptions, Ceph and nooba components or not(Check all 7 subscriptions).
+        3. Apply custom taint to all nodes.
+        4. Verify the pods in all nodes are running as per taints applied.
+        5. Reboot one of the nodes and check toleration on all odf pods on that node.
+        7.Replace one of the nodes and check all odf pods on that node are running.
```
Comment (Shrivaibavi) [inline tests/functional/z_cluster/nodes/test_non_ocs_taint_and_toleration.py:242]: are we missing 6th step here ??

File: tests/functional/z_cluster/nodes/test_non_ocs_taint_and_toleration.py [hunk 2]
```diff
@@ -227,3 +228,104 @@ def test_non_ocs_taint_and_tolerations(self):
...
+            for pod in pod_list:
+                pod.delete(wait=False)
+
+        logger.info("After edit noticed few pod respins as expected")
+        assert wait_for_pods_to_be_running(timeout=1200, sleep=15)
+
+        logger.info(
+            "Check custom toleration on all newly created pods under openshift-storage"
+        )
+        assert not check_toleration_on_pods(toleration_key="xyz")
```
Comment (Shrivaibavi) [inline tests/functional/z_cluster/nodes/test_non_ocs_taint_and_toleration.py:331]: I dont see reboot and replace of nodes as per the test steps, are we missing something here ??

File: tests/functional/z_cluster/nodes/test_non_ocs_taint_and_toleration.py [hunk 3]
```diff
@@ -227,3 +228,104 @@ def test_non_ocs_taint_and_tolerations(self):
                 resource_count=count * replica_count,
             ), "New OSDs failed to reach running state"
             check_ceph_health_after_add_capacity(ceph_rebalance_timeout=2500)
+
+    @skipif_ocs_version("<=4.15")
+    def test_custom_taint_node_operations(self, nodes):
+        """
+        Test runs the following steps
```
Comment (avd-sagare) [inline tests/functional/z_cluster/nodes/test_non_ocs_taint_and_toleration.py:235]: what are negative tests here?

File: tests/functional/z_cluster/nodes/test_non_ocs_taint_and_toleration.py [hunk 4]
```diff
@@ -227,3 +228,104 @@ def test_non_ocs_taint_and_tolerations(self):
                 resource_count=count * replica_count,
             ), "New OSDs failed to reach running state"
             check_ceph_health_after_add_capacity(ceph_rebalance_timeout=2500)
+
```
Comment (avd-sagare) [inline tests/functional/z_cluster/nodes/test_non_ocs_taint_and_toleration.py:231]: Add Polarioan id

File: tests/functional/z_cluster/nodes/test_non_ocs_taint_and_toleration.py [hunk 5]
```diff
@@ -227,3 +228,104 @@ def test_non_ocs_taint_and_tolerations(self):
...
+        logger.info("Check custom toleration on all subscriptions")
+
+        assert not check_toleration_on_subscriptions(toleration_key="xyz")
+
+        if config.ENV_DATA["mcg_only_deployment"]:
+            logger.info("Wait some time after adding toleration for pods respin")
+            waiting_time = 60
+            logger.info(f"Waiting {waiting_time} seconds...")
+            time.sleep(waiting_time)
```
Comment (avd-sagare) [inline tests/functional/z_cluster/nodes/test_non_ocs_taint_and_toleration.py:316]: Instead of sleep you  use timeout sampler

**Pull request closed without merging**